src/app/pages/account/users/user-form/user-form.component.spec.ts

```diff
@@ -0,0 +1,279 @@
+
+      expect(values).toEqual({
+        'Auxiliary Groups': ['test-group'],
+        'Confirm Password': '',
+        'Create New Primary Group': false,
+        'Disable Password': false,
+        'Full Name': 'test',
+        'Home Directory Permissions': '700',
+        'Home Directory': '/home/test',
+        'Lock User': false,
+        'Microsoft Account': false,
+        'Permit Sudo': false,
+        'Primary Group': 'test-group',
+        'Samba Authentication': true,
+        'SSH Public Key': '',
+        UID: '1004',
+        Email: '',
+        Password: '',
+        Shell: 'bash',
+        Username: 'test',
+      });
+    });
+
+    it('sends an update payload to websocket and closes modal when save is pressed', async () => {
+      const form = await loader.getHarness(IxFormHarness);
+      await form.fillForm({
+        'Auxiliary Groups': ['mock-group'],
+        'Full Name': 'updated',
+        'Home Directory Permissions': '755',
+        'Home Directory': '/home/updated',
+        'Microsoft Account': true,
+        'Permit Sudo': true,
+        'Primary Group': 'mock-group',
+        'Samba Authentication': false,
+        'Lock User': true,
+        Shell: 'zsh',
+        Username: 'updated',
+      });
+
+      const saveButton = await loader.getHarness(MatButtonHarness.with({ text: 'Save' }));
+      await saveButton.click();
+
+      expect(ws.call).toHaveBeenCalledWith('user.update', [
+        69,
+        {
+          email: null,
+          full_name: 'updated',
+          group: 102,
+          groups: [102],
+          home_mode: '755',
+          home: '/home/updated',
+          locked: true,
+          microsoft_account: true,
+          password_disabled: false,
+          shell: '/usr/bin/zsh',
+          smb: false,
+          sshpubkey: null,
+          sudo: true,
+          username: 'updated',
+        },
+      ]);
+    });
+  });
+
+  describe('checks form states', () => {
+    it('set disable password is true and check inputs', async () => {
+      spectator.component.setupForm();
+
+      const form = await loader.getHarness(IxFormHarness);
+      form.fillForm({
+        'Disable Password': true,
+      });
+
+      const disabled = await form.getDisabledState();
+      expect(disabled).toEqual(expect.objectContaining({
+        'Confirm Password': true,
+        'Lock User': true,
+        'Permit Sudo': true,
+        Password: true,
+      }));
+    });
+
+    it('check form inputs when user is builtin', async () => {
+      spectator.component.setupForm({ ...mockUser, builtin: true });
+
+      const form = await loader.getHarness(IxFormHarness);
+      const disabled = await form.getDisabledState();
+
+      expect(disabled).toEqual(expect.objectContaining({
+        'Home Directory Permissions': true,
+        'Home Directory': true,
+        'Primary Group': true,
+        Username: true,
+      }));
+    });
+  });
+
+  it('checks download ssh public key button', async () => {
```

Comment:
Here and in other places I think it would better to describe test cases from the standpoint of how system works, i.e. `it('only shows download ssh key button when ssh key is filled in` or something similar.